plugins/rest/azure.go

```diff
@@ -0,0 +1,138 @@
+package rest
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+var (
+	azureIMDSEndpoint = "http://169.254.169.254/metadata/identity/oauth2/token"
```

Comment:
I wonder if we should have defaults for `api-version` & `resource` as well. `api-version` changes but not often and the `resource` will rarely change. Giving the option to change is good but good defaults may be helpful.

Reply:
It seems `api-version` would logically have `"2018-02-01"` as a default version, so I'll go ahead and add that. I'm not too sure about the `resource` field though. In your ticket the resource is https://storage.azure.com/, but the [example](https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/how-to-use-vm-token#get-a-token-using-http) uses https://management.azure.com/. I'm not familiar with Azure's managed identities so perhaps one of these is clearly suitable for a default. In that case I'd be happy to add it. What do you suggest?

Reply:
You are right that the mechanism in place can be used to get access token for different resources. However, this specific use case is for fetching OPA policies & data bundles from the Azure storage account. So using https://storage.azure.com/ seems like a good default to me. I doubt if anyone would access any other resource from this plugin. Plus this is a default value and not the final value that cannot be overridden.

Reply:
That's enough to convince me. Will make the change. Resolving.